internal/infraestructure/github-connection/github-connection.go

```diff
@@ -2,54 +2,185 @@ package github_connection
 }
 
-func NewGithubConnection(params GithubConnectionParams) (*GithubConnection, error) {
+type FileChange struct {
+	github.CommitFile
+	Position int `json:"position"`
+}
+
+func NewGithubConnection() (*GithubConnection, error) {
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		return nil, fmt.Errorf("GITHUB_TOKEN environment variable is not set")
+	}
+
+	repoFullName := os.Getenv("GITHUB_REPOSITORY")
+	if repoFullName == "" {
+		return nil, fmt.Errorf("GITHUB_REPOSITORY environment variable is not set")
+	}
+	repoParts := strings.Split(repoFullName, "/")
+	if len(repoParts) != 2 {
+		return nil, fmt.Errorf("invalid GITHUB_REPOSITORY format: %s", repoFullName)
+	}
+	repoOwner := repoParts[0]
+	githubRepositoryName := repoParts[1]
+
+	pullRequestNumberStr := os.Getenv("GITHUB_PR_NUMBER")
+	if pullRequestNumberStr == "" {
+		return nil, fmt.Errorf("GITHUB_PR_NUMBER environment variable is not set")
+	}
+	pullRequestNumber, err := strconv.Atoi(pullRequestNumberStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid GITHUB_PR_NUMBER: %w", err)
+	}
 
-	GithubClient := github.NewClient(nil).WithAuthToken(params.Token)
+	eventName := os.Getenv("GITHUB_EVENT_NAME")
+	if eventName == "" {
+		return nil, fmt.Errorf("GITHUB_EVENT_NAME environment variable is not set")
+	}
+	GithubClient := github.NewClient(nil).WithAuthToken(token)
 
 	return &GithubConnection{
-		client:            GithubClient,
-		RepositoryName:    params.RepositoryName,
-		RepoOwner:         params.RepoOwner,
-		PullRequestNumber: params.PullRequestNumber,
+		Client:            GithubClient,
+		RepositoryName:    githubRepositoryName,
+		RepoOwner:         repoOwner,
+		PullRequestNumber: pullRequestNumber,
+		EventName:         eventName,
 	}, nil
 }
 
-type GithubConnection struct {
-	client            *github.Client
-	RepositoryName    string
-	RepoOwner         string
-	PullRequestNumber int
+func (receiver *GithubConnection) GetEventName() string {
+	return receiver.EventName
 }
 
-func (receiver *GithubConnection) GetRepositoryName() string {
+func (receiver *GithubConnection) GetRepository() string {
 	return receiver.RepositoryName
 }
 
-func (receiver *GithubConnection) GetRepository() string {
-	return receiver.RepositoryName
+func (receiver *GithubConnection) GetPullRequestChanges() ([]github.CommitFile, error) {
+	var ctx = context.Background()
+	var opt = &github.ListOptions{PerPage: 100}
+	var allChanges []github.CommitFile
+
+	for {
+		files, resp, err := receiver.Client.PullRequests.ListFiles(
+			ctx, receiver.RepoOwner,
+			receiver.RepositoryName,
+			receiver.PullRequestNumber,
+			opt)
+
+		if err != nil {
+			return nil, fmt.Errorf("error listing pull request files: %w", err)
+		}
+		for _, file := range files {
+			if file.GetAdditions() == 0 {
+				continue
+			}
+			allChanges = append(allChanges, *file)
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+
+	return allChanges, nil
 }
```

Comment:
Code Review 2024-08-18 02:28:52 - New lines 56 to 84:

func (receiver *GithubConnection) GetPullRequestChanges() ([]github.CommitFile, error) {
	var ctx = context.Background()
	var opt = &github.ListOptions{PerPage: 100}
	var allChanges []github.CommitFile

	for {
		files, resp, err := receiver.Client.PullRequests.ListFiles(
			ctx, receiver.RepoOwner,
			receiver.RepositoryName,
			receiver.PullRequestNumber,
			opt)

		if err != nil {
			return nil, fmt.Errorf("error listing pull request files: %w", err)
		}
		for _, file := range files {
			if file.GetAdditions() == 0 {
				continue
			}
			allChanges = append(allChanges, *file)
		}

		if resp.NextPage == 0 {
			break
		}
		opt.Page = resp.NextPage
	}

	return allChanges, nil